rust/hyperlane-core/src/config/config_path.rs

```diff
@@ -0,0 +1,89 @@
+use std::fmt::{Display, Formatter};
+use std::ops::Add;
+use std::sync::Arc;
+
+use convert_case::{Case, Casing};
+use itertools::Itertools;
+
+/// Path within a config tree.
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+pub struct ConfigPath(Vec<Arc<String>>);
+
+impl Display for ConfigPath {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.json_name())
+    }
+}
+
+impl<S: Into<String>> Add<S> for &ConfigPath {
+    type Output = ConfigPath;
+
+    fn add(self, rhs: S) -> Self::Output {
+        self.join(rhs)
+    }
+}
+
+impl Add<ConfigPath> for &ConfigPath {
+    type Output = ConfigPath;
+
+    fn add(self, rhs: ConfigPath) -> Self::Output {
+        self.merge(&rhs)
+    }
+}
+
+impl ConfigPath {
+    /// Add a new part to the path.
+    pub fn join(&self, part: impl Into<String>) -> Self {
+        let part = part.into();
+        debug_assert!(!part.contains('.'));
```

Comment:
wondering what situation this is supposed to catch and if it should be something we want in the optimized build?

(i know this is just a moved bit of code that already existed)

Reply:
It catches a case that isn't going to break anything, but that would be nice to catch during debug. Basically we should not have the path include "." but have different segments instead.